File: loops/notes.py
# Julia Properzi,, Loops notes python

#What is a loop? 
    #a section of code that repeats multiple times
#What are the two types of loops?
    #for loop
nums = [12,3,66,7,3,3,2]

for num in nums: 
    print(num)

    #While loop
    x = 0

while x < 10:
    print(x)
    x+=1
#What is iteration?
    #one particular instance of the loop
    #iteration the amount of times you are looping through

#What are lists? 
    #a variable that holds multiple values
nums = [1,2,3,4,5,6,7,6]
siblings = ["Alex","Katie", "Andrew", "Vienna", "Tia", "Treyson", "Xavier", "Hailey"]
print(nums) #prints whole list is ugly
print(siblings[3]) #prints 1 item from the list

siblings[7] = "Jake" #can replace the name in the list
siblings.pop(5) #removes something from the list
siblings.insert(1,"Jayshree") #Can add to the list
siblings.insert(2, ["Joe", "Noah", "Zee"]) #can put list in list
print(siblings) 

#How do you make lists in python? 
#To make a list, step one is brackets
#step two you add in items with correct data types
#Step three, you need to have a comma between each item

#How do you make for loops in python? (use plural)
for sibling in siblings:
    print(sibling)

for x in range(1,101, 20):
    print(x)
#How do you make while loops in python?
import random 
x = 1 #variable to keep count of iteration
goose = random.randint(1,20)

while x < 20:
    if x == goose:
        print("goose!")
        break #tells loop to stop
    #continue sends back to while and starts infinate loop
    else:
        print("duck")
    x+=1

    #continue moves on to the next iteration without finishing

File: loops/notes.c
// Julia Properzi, loops notes in C
#include <stdio.h>

int main(void){
    //What is a loop? A section of code repeated multiple times
    //What are the two types of loops?
    //while loops
    int start = 0;
    while (start< 5){
        printf("Hello\n");
        start++;
    }
    //for loop
    int q;
    for(q=0;q<5;q++){
        printf("%d\n", q);
    }
    //What is iteration? Repeating something with minor changes each time
    //What are arrays(list)? an array is a variable holding multiple values (same thing just different word in C)
    //How do you make arrays(lists) in C? array items must all be the same data type!
    int grades[] = {88, 97, 100, 70, 72, 99, 61};
    // one set data type, then AFTER naming put brackets and write the length of the list, then the list is surrounded by curly brackets {}, finally commans between each item.
    //  print a single item from a list
    printf("CSP Grade: %d\n", grades[2]);
    // change an item in the array
    grades[2]= 95;
    printf("CSP Grade: %d\n", grades[2]);
    // size of array(list) in bytes
    int size = sizeof(grades);
    //length in arrays(list) items
    int length = sizeof(grades)/sizeof(grades[0]);
    printf("The array is %d items long.\n", length);
    //array with strings
    //first brackets sets length of array(lists)
    //second brackets set the length of each string
    char movies[][20] = {"Cars", "Treasure Planet", "An American Tale", "Marley and Me", "The Avengers"};
    printf("The movie is %s!\n", movies[1]);
    int mlength = sizeof (movies)/ sizeof(movies[0]);
    //How do you make for loops in C?
    // set the iterator, keeps track of times through the loop (best practice to set as x or i)
    int x;
    // in parens (starting point; ending point; count by)
    for(x=0;x<=10;x+=2){
        printf("%d\n", x);
    }
    int m;
    for(m=0;m<mlength;m++){
        printf("%s \n", movies[m]);
    }
    //How do you make while loops in C?
    int w = 0;

    while(w>=0){// stop point
        printf("%d\n", w);
        w -= 10;//what we count by
    }
    
    return 0;
}